build/source/engine/read_attrb.f90

```diff
@@ -55,27 +61,44 @@ USE get_ixname_module,only:get_ixAttr,get_ixType  ! access function to find inde
+ integer(i4b)                         :: varid           ! netcdf variable id
+ character(LEN=nf90_max_name)         :: var_name        ! character array of netcdf variable name
  integer(i4b),parameter               :: categorical=101 ! named variable to denote categorical data
  integer(i4b),parameter               :: numerical=102   ! named variable to denote numerical data
  integer(i4b),allocatable             :: varType(:)      ! type of variable (categorical or numerical)
  integer(i4b),allocatable             :: varIndx(:)      ! index of variable within its data structure
+ integer(i4b)                         :: nvar            ! number of variables in netcdf local attribute file
  integer(i4b)                         :: iAtt            ! index of an attribute name
  integer(i4b)                         :: iHRU            ! index of an HRU
+ integer(i4b)                         :: iGRU            ! index of an GRU
  integer(i4b)                         :: nAtt            ! number of model attributes
  integer(i4b)                         :: nVar_attr       ! number of variables in the model attribute structure
  integer(i4b)                         :: nVar_type       ! number of variables in the model category structure
  logical(lgt),allocatable             :: checkType(:)    ! vector to check if we have all desired categorical values
  logical(lgt),allocatable             :: checkAttr(:)    ! vector to check if we have all desired local attributes
+ ! define mapping variables
+ integer(i4b)                         :: ix_gru             ! index of current gru
+ integer(i4b)                         :: ix_hru             ! index of current hru
+ integer(i4b),allocatable             :: gru_ix(:)          ! sequential index of GRUs
+ integer(i4b),allocatable             :: hru_ix(:)          ! sequential index of hru over entire domain
+ integer(i4b),allocatable             :: hru2gru_ixGRU(:)   ! sequential index of GRU for each HRU
+ integer(i4b),allocatable             :: hru2gru_ixHRU(:)   ! squential index of HRUs in each GRUs
+ integer(i4b),allocatable             :: hru_id(:)          ! unique id of hru over entire domain
+ integer(i4b),allocatable             :: gru_id(:)          ! unique ids of GRUs
+ integer(i4b),allocatable             :: hru2gru_id(:)      ! unique GRU ids at each HRU
+ integer(i4b),allocatable             :: categorical_var(:) ! temporary categorical variable from local attributes netcdf file
+ real(dp),allocatable                 :: numeric_var(:)     ! temporary numeric variable from local attributes netcdf file
+ ! define indicators for different data structures
+ integer(i4b),parameter               :: ixType=1           ! indicator for type_gru structure
+ integer(i4b),parameter               :: ixAttr=2           ! indicator for attr_gru structure 
+ integer(i4b), parameter              :: imiss = -999       ! missing value for variable id
```

Comment:
Here as above. Seems like missing values should be synchronized over entire program.